python_modules/dagster/dagster/_core/execution/plan/inputs.py

```diff
@@ -973,33 +883,12 @@ def required_resource_keys(self, _job_def: JobDefinition) -> Set[str]:
 
 
 @whitelist_for_serdes(storage_field_names={"node_handle": "solid_handle"})
-class FromDynamicCollect(
-    NamedTuple(
-        "_FromDynamicCollect",
-        [
-            ("source", Union[FromPendingDynamicStepOutput, FromUnresolvedStepOutput]),
-            # deprecated, preserved for back-compat
-            ("node_handle", NodeHandle),
-            ("input_name", str),
-        ],
-    ),
-):
-    def __new__(
-        cls,
-        source: Union[FromPendingDynamicStepOutput, FromUnresolvedStepOutput],
-        # deprecated, preserved for back-compat
-        node_handle: Optional[NodeHandle] = None,
-        input_name: Optional[str] = None,
-    ):
-        return super().__new__(
-            cls,
-            source=source,
-            # add placeholder values for back-compat
-            node_handle=check.opt_inst_param(
-                node_handle, "node_handle", NodeHandle, default=NodeHandle("", None)
-            ),
-            input_name=check.opt_str_param(input_name, "input_handle", default=""),
-        )
+@record
+class FromDynamicCollect:
+    source: Union[FromPendingDynamicStepOutput, FromUnresolvedStepOutput]
+    # deprecated, preserved for back-compat
+    node_handle: NodeHandle = NodeHandle("", None)
+    input_name: str = ""
```

Comment:
looks like this `node_handle` and `input_name` get the same backcompat treatment owen pointed out